modules/internal/OwnedObject.chpl

```diff
@@ -199,39 +199,51 @@ module OwnedObject {
        refer to `nil` after this call.
      */
     proc init=(pragma "leaves arg nil" pragma "nil from arg" ref src:_owned) {
-      if isNonNilableClass(this.type) && isNilableClass(src) then
-        compilerError("cannot create a non-nilable owned variable from a nilable class instance");
+      // Use 'this.type.chpl_t' if it is set in case RHS is a subtype
+      this.chpl_t = if this.type.chpl_t != ?
```

Comment:
Here's something I didn't know you could do!

Reply:
Looks like we already had some of these, but using the reverse of the conditional. Could probably be a helper function.